preprocessor/Boundary_Conditions_From_Options.F90

```diff
@@ -1187,6 +1193,26 @@ subroutine set_vector_boundary_conditions_values(state, field, bc_path, position
               scalar_surface_field => extract_scalar_surface_field(field, bc_name, name="WettingDryingAlpha")
               call zero(scalar_surface_field)
            end if
+           if (have_option(trim(bc_path_i)//"/external_density")) then
+              call get_boundary_condition(field, i+1, surface_mesh=surface_mesh, &
+                 surface_element_list=surface_element_list)
+              scalar_surface_field => extract_scalar_surface_field(field, bc_name, name="ExternalDensity")
+              if (have_option(trim(bc_path_i)//"/external_density/from_field")) then
+                ! from_field: The parent field contains the boundary values that you want to apply to surface_field.
+                call get_option(trim(bc_path_i)//"/external_density/from_field/parent_field_name", parent_field_name)
+                scalar_parent_field => extract_scalar_field(state, parent_field_name, stat)
+                if(stat /= 0) then
+                  FLExit("Could not extract parent field. Check options file?")
```

Comment:
Perhaps worth adding to the error here "parent *scalar* field" as you've only checked for that (understandably) but the schema documentation (because its been reused from other purposes) is unclear about this.